### PR TITLE
EUI-8821 EUI-8731 Register Other Org - Organisation services amendments

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-8821
https://tools.hmcts.net/jira/browse/EUI-8731

### Change description ###
EUI-8821 Register other org - (Display & Functionality) for "Service not listed" on Which services will your organisation need to access page
EUI-8731 Register other org - (EUI LoV) List of services accessed

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts [hunk 1]
```diff
@@ -1,11 +1,8 @@
...
-import { FormControl, FormGroup } from '@angular/forms';
-import { ActivatedRoute, Router } from '@angular/router';
-import { Subscription } from 'rxjs';
-import { OrganisationServicesMessage } from '../../../register-org/models';
-import { LovRefDataModel } from '../../../shared/models/lovRefData.model';
-import { LovRefDataService } from '../../../shared/services/lov-ref-data.service';
-import { SERVICES_REF_DATA } from '../../__mocks__';
+import { FormArray, FormControl, FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+import { ORGANISATION_SERVICES } from 'src/register-org/constants/register-org-constants';
```
Comment (connorpgpmcelroy) [inline src/register-org/components/organisation-services-access/organisation-services-access.component.ts:4]: src used - minor issue

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Minor issue

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/register-org/components/organisation-services-access/organisation-services-access.component.ts]: I have applied the fix now.

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Review by udaydenduluri33 — APPROVED** [review]

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts [hunk 2]
```diff
@@ -15,65 +12,58 @@ import { RegisterOrgService } from '../../services/register-org.service';
...
-    }
   }
 
   public onServicesSelectionChange(event: any): void {
     if (event.target.checked) {
-      this.services.find((service) => service.value_en === event.target.value).selected = true;
+      this.selectedServices.push(event.target.value);
     } else {
-      this.services.find((service) => service.value_en === event.target.value).selected = false;
+      const serviceIndex = this.selectedServices.findIndex((service) => service === event.target.value);
```
Comment (Daniel-Lam) [inline src/register-org/components/organisation-services-access/organisation-services-access.component.ts:51]: @johnbenjamin-hmcts - does this still work if `findIndex()` returns -1?

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts [hunk 3]
```diff
@@ -87,21 +77,38 @@ export class OrganisationServicesAccessComponent extends RegisterComponent imple
...
     this.services.forEach((service) => {
-      if (this.registrationData.services.includes(service.value_en)) {
-        service.selected = true;
-      }
+      service.selected = this.selectedServices.includes(service.key);
     });
+    if (this.registrationData.otherServices) {
+      this.showOtherServicesInput = true;
+      this.selectedServices.push('NONE');
+      this.services.find((service) => service.key === 'NONE').selected = true;
```
Comment (Daniel-Lam) [inline src/register-org/components/organisation-services-access/organisation-services-access.component.ts:88]: @johnbenjamin-hmcts - is `services.find()` always guaranteed to return a truthy result?

File: src/register-org/components/check-your-answers/check-your-answers.component.ts [hunk 4]
```diff
@@ -13,19 +14,28 @@ export class CheckYourAnswersComponent extends RegisterComponent implements OnIn
...
 
     this.cyaFormGroup = new FormGroup({
       confirmTermsAndConditions: new FormControl(null, Validators.required)
     });
-  }
 
-  public ngOnInit(): void {
-    super.ngOnInit();
+    this.registrationData.services?.forEach((serviceKey) => {
+      const service = ORGANISATION_SERVICES.find((service) => service.key === serviceKey).value;
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.ts:33]: @johnbenjamin-hmcts - is `ORGANISATION_SERVICES.find()` always guaranteed to return a truthy result?

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just a few minor things to look at

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply src/register-org/components/organisation-services-access/organisation-services-access.component.ts:51]: @Daniel-Lam `ORGANISATION_SERVICES` is a hard-coded master list of services that we display in the page, so the `selectedServices` must always contain the services from this master list.

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply src/register-org/components/organisation-services-access/organisation-services-access.component.ts:88]: `NONE` is always a part of the master hard-coded list of data `ORGANISATION_SERVICES`. So, the `find` will always return a result.

File: src/register-org/components/check-your-answers/check-your-answers.component.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply src/register-org/components/check-your-answers/check-your-answers.component.ts:33]: `ORGANISATION_SERVICES` is the master list of hard-coded services to display, and also I am checking for empty services in the `forEach` loop.

File: src/register-org/components/organisation-services-access/organisation-services-access.component.ts [hunk 5]
```diff
@@ -87,21 +77,38 @@ export class OrganisationServicesAccessComponent extends RegisterComponent imple
...
+    this.otherServicesError = null;
+    if (!this.selectedServices.length && !this.showOtherServicesInput) {
       this.validationErrors.push({
         id: this.services[0].key,
         message: OrganisationServicesMessage.NO_ORG_SERVICES
       });
+      this.noServicesError = OrganisationServicesMessage.NO_ORG_SERVICES;
+    }
+    if (this.showOtherServicesInput && !this.servicesFormGroup.get('otherServices').value) {
+      console.log('SERVICES', this.services);
```
[the remark this answers is not in the document]
Comment (johnbenjamin-hmcts) [reply src/register-org/components/organisation-services-access/organisation-services-access.component.ts:105]: @Daniel-Lam - I will handle this as part of https://github.com/hmcts/rpx-xui-manage-organisations/pull/1113 which is the next ticket I am working on.

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**